synapseclient/models/team.py

```diff
@@ -56,6 +57,80 @@ def fill_from_dict(
         return self
 
 
+@dataclass
+class TeamMembershipStatus:
+    """
+    Contains information about a user's membership status in a Team.
+    Represents a [Synapse TeamMembershipStatus](<https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/TeamMembershipStatus.html>).
+    User definable fields are:
+
+    Attributes:
+        team_id: The id of the Team.
+        user_id: The principal id of the user.
+        is_member: true if and only if the user is a member of the team
+        has_open_invitation: true if and only if the user has an open invitation to join the team
+        has_open_request: true if and only if the user has an open request to join the team
+        can_join: true if and only if the user requesting this status information can join the user to the team
+        membership_approval_required: true if and only if team admin approval is required for the user to join the team
+        has_unmet_access_requirement: true if and only if there is at least one unmet access requirement for the user on the team
+        can_send_email: true if and only if the user can send an email to the team
+    """
+
+    team_id: Optional[str] = None
+    """The ID of the team"""
+
+    user_id: Optional[str] = None
+    """The ID of the user"""
```

Comment:
On the https://github.com/Sage-Bionetworks/synapsePythonClient/blob/dffef121d550630c4e4161475917417178192da3/synapseclient/models/user.py#L110-L111 object the ID is an int. Is there a reason to use a string instead here, or could we keep this consistent? This would have some changes across a few lines in the PR to make it a consistent int if changing it.

Reply:
Based on the synapse documentation [here](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/TeamMembershipStatus.html), `user_id` and `team_id` are both string. 

I also looked into the `UserProfile`: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/UserProfile.html, and it looks like that `ownerId` which is basically the same as `id` in UserProfile is also a string. 

Should they all be kept as string to align with the synapse documentation? If not, I could make `team_id` and `user_id` integers.

Reply:
Great points. this is fine as is